### DISCUSSION TITLE
How to use peft for LoRA ensemble

### DISCUSSION THREAD

**Discussion by DongxiaW** [huggingface/peft#1945] [Q&A] [answered]
Can we use peft for LoRA ensemble? i.e., train multiple LoRA adapters and generate one output for each adapter in a single forward pass?

**Comment by BenjaminBossan** [comment, marked as answer]
No, this is not possible. You can load multiple LoRA adapters and then cycle through each of them to get the same outcome, but it will be more expensive to call. What you describe cannot be easily achieved as that would require to change the output signature of each module's `forward` method.
